### PR TITLE
Search closest enabled date on init of the dxCalendar (T701103)

### PR DESCRIPTION
**Pull request by mrjohn0011**

### CODE REVIEW

File: js/ui/calendar/ui.calendar.js [hunk 1]
```diff
@@ -344,50 +345,81 @@ var Calendar = Editor.inherit({
...
     },
 
+    _normalizeOffset: function(zoomLevel, offset) {
+        var offsetCorrection = 2 * offset / Math.abs(offset);
+
+        if(Math.abs(offset) <= 1) {
+            return offset;
+        }
+
+        if(zoomLevel === "decade") {
```
Comment (DokaRus) [inline js/ui/calendar/ui.calendar.js]: How do you think about replacing `zoomLevel` values with constants?

File: testing/tests/DevExpress.ui.widgets.editors/calendar.tests.js [hunk 2]
```diff
@@ -1,28 +1,28 @@
...
-    browser = require("core/utils/browser"),
-    dataUtils = require("core/element_data"),
-    dateLocalization = require("localization/date");
-
-var camelize = inflector.camelize;
-
-require("common.css!");
-require("generic_light.css!");
+import $ from "jquery";
+import inflector from "core/utils/inflector";
```
Comment (DokaRus) [inline testing/tests/DevExpress.ui.widgets.editors/calendar.tests.js]:
```suggestion
 import { camelize } from "core/utils/inflector";
```

File: testing/tests/DevExpress.ui.widgets.editors/calendar.tests.js [hunk 3]
```diff
@@ -1,28 +1,28 @@
...
+import Views from "ui/calendar/ui.calendar.views";
+import Calendar from "ui/calendar";
+import pointerMock from "../../helpers/pointerMock.js";
+import keyboardMock from "../../helpers/keyboardMock.js";
+import config from "core/config";
+import browser from "core/utils/browser";
+import dataUtils from "core/element_data";
+import dateLocalization from "localization/date";
+
+const camelize = inflector.camelize;
```
Comment (DokaRus) [inline testing/tests/DevExpress.ui.widgets.editors/calendar.tests.js]:
```suggestion

```

**Review by DokaRus — DISMISSED** [review]

File: js/ui/calendar/ui.calendar.js [hunk 4]
```diff
@@ -37,6 +38,11 @@ var CALENDAR_CLASS = "dx-calendar",
     CALENDAR_INPUT_STANDARD_PATTERN = "yyyy-MM-dd",
     CALENDAR_DATE_VALUE_KEY = "dxDateValueKey",
 
+    ZOOM_LEVEL_MONTH = "month",
```
Comment (churkin) [inline js/ui/calendar/ui.calendar.js]:
```javascript
const zoolLevel = {
   MONTH: "month",
   YEAR: "year",
   ...
}
```

File: js/ui/calendar/ui.calendar.js [hunk 5]
```diff
@@ -344,50 +350,81 @@ var Calendar = Editor.inherit({
         this.option(optionName, dateSerialization.serializeDate(optionValue, serializationFormat));
     },
 
+    _normalizeOffset: function(zoomLevel, offset) {
+        var offsetCorrection = 2 * offset / Math.abs(offset);
+
+        if(Math.abs(offset) <= 1) {
```
Comment (churkin) [inline js/ui/calendar/ui.calendar.js]: extract `Math.abs(offset)` to a var

File: js/ui/calendar/ui.calendar.js [hunk 6]
```diff
@@ -344,50 +350,81 @@ var Calendar = Editor.inherit({
...
         var currentDate = baseDate || new Date(this.option("currentDate")),
-            maxDate = this.option("max"),
-            minDate = this.option("min"),
+            maxDate = this._getMaxDate(),
+            minDate = this._getMinDate(),
             zoomLevel = this.option("zoomLevel"),
-            isAvailableDateFound = false,
-            newDate;
+            dateForward = new Date(currentDate),
+            dateBackward = new Date(currentDate),
```
Comment (churkin) [inline js/ui/calendar/ui.calendar.js:362]: `dateForwardInRange === dateBackwardInRange` ?

**Review by churkin — CHANGES_REQUESTED** [review]

**Comment by mrjohn0011** [comment]
Close because of incorrect rebase. This PR will be recreated

**Pull request closed without merging**